cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/state/AppState.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.internal.app.store.state;
+
+import javax.annotation.Nullable;
+
+/**
+ * Represents an application state.
+ */
+public class AppState {
+  private final String namespace;
+  private final String appName;
+  private final String stateKey;
+  private final byte[] stateValue;
+
+  public AppState(String namespace,
+                  String appName) {
+    this(namespace, appName, null, null);
+  }
+
+  public AppState(String namespace,
+                  String appName,
+                  String stateKey) {
+    this(namespace, appName, stateKey, null);
+  }
+
+  public AppState(String namespace,
+                  String appName,
+                  @Nullable String stateKey,
+                  @Nullable byte[] stateValue) {
```

Comment:
I don't think we want to allow null values, as the semantics become kind of unclear. For example, is writing a null value the same as deleting the key? Or do we need to differentiate between a null and not exists? Simpler just not to allow it and return a 400 if somebody tries to do it.

I saw this class is also being used when fetching the value for a key, which I would guess is another reason this is nullable. For that case, we should be using a different class. For example, create an AppStateKey class with namespace, app, and key. Then rename this to AppStateKeyValue that extends AppStateKey and adds a value variable.